app/services/data_migrations/backfill_withdrawn_or_declined_for_candidate_by_provider.rb

```diff
@@ -0,0 +1,29 @@
+module DataMigrations
+  class BackfillWithdrawnOrDeclinedForCandidateByProvider
+    TIMESTAMP = 20220120161816
+    MANUAL_RUN = false
+
+    def change
+      audits_joins_sql = <<-SQL.squish
+      INNER JOIN audits ON audits.auditable_type = 'ApplicationChoice'
+      AND audits.auditable_id = application_choices.id
+      AND audits.user_type = 'ProviderUser'
+      AND audits.comment IN ('Declined on behalf of the candidate', 'Withdrawn on behalf of the candidate')
+      SQL
+
+      withdrawn_or_declined_by_provider_applications = ApplicationChoice.joins(audits_joins_sql).where('application_choices.status': %w[declined withdrawn]).distinct
```

Comment:
can you test those queries on prod manually to ensure they don't timeout as the audits table is a bit dodgy these days?

Reply:
We did run it on prod earlier, something like 828 existing withdrawn/declined by provider application choices. It seemed relatively fast as well!